src/lib/mina_base/transaction_logic.ml

```diff
@@ -1129,18 +1137,21 @@ module Make (L : Ledger_intf) : S with type ledger := L.t = struct
     let%bind delegate =
       if Token_id.(equal default) a.token_id then
         update a.permissions.set_delegate delegate a.delegate
-          ~is_keep:(( = ) Set_or_keep.Keep) ~update:(fun u x ->
-            match u with Keep -> x | Set y -> Some y )
+          ~is_keep:((Set_or_keep.equal Poly.( = )) Set_or_keep.Keep)
```

Comment:
Do these need to be polymorphic equality? If so, can you add a comment explaining why?

Reply:
Replaced these with a dedicated `Set_or_keep.is_keep` function.